components/com_content/src/Service/Router.php

```diff
@@ -274,6 +283,27 @@ public function getArticleId($segment, $query)
 			return (int) $this->db->loadResult();
 		}
 
+		if ($this->legacyStrict)
+		{
+			$id    = (int) $segment;
+			$alias = ltrim(str_replace($id, '', $segment), '-');
+
+			$dbquery = $this->db->getQuery(true);
+			$dbquery->select($this->db->quoteName('id'))
+				->from($this->db->quoteName('#__content'))
+				->where(
+					[
+						$this->db->quoteName('alias') . ' = :alias',
+						$this->db->quoteName('id') . ' = :id',
```

Comment:
```suggestion
						$this->db->quoteName('id') . ' = :id',
						$this->db->quoteName('alias') . ' = :alias',
```

Maybe not so important but it's strange if it different to the bind statement